src/pages/test-report/components/competency/competency.ts

```diff
@@ -1,41 +1,91 @@
 import { Component, Input, ViewChild, ElementRef, Renderer2 } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
+import { merge } from 'rxjs/observable/merge';
+import { map } from 'rxjs/operators';
+
+import { StoreModel } from '../../../../shared/models/store.model';
+import { AddDrivingFault } from '../../../../modules/tests/test_data/test-data.actions';
 import { HammerProvider } from '../../../../providers/hammer/hammer';
+import { Competencies } from '../../../../modules/tests/test_data/test-data.constants';
+import { competencyLabels } from './competency.constants';
+import { getCurrentTest } from '../../../../modules/tests/tests.selector';
+import { getTestData } from '../../../../modules/tests/test_data/test-data.reducer';
+import { getDrivingFaultCount } from '../../../../modules/tests/test_data/test-data.selector';
 
 enum CssClassesEnum {
   DRIVING_FAULT = 'driving-fault',
   RIPPLE_EFFECT = 'ripple-effect',
 }
 
+interface ComptencyState {
+  drivingFaultCount$: Observable<number>;
+}
+
 @Component({
   selector: 'competency',
   templateUrl: 'competency.html',
   providers: [HammerProvider],
 })
 export class CompetencyComponent {
 
-  @Input() label: string;
-  rippleEffectAnimationDuration: number = 300;
-  // TODO - This needs to be gotten from the store
-  faultCount: number = 0;
+  @Input()
+  competency: Competencies;
 
   @ViewChild('competencyButton')
   button: ElementRef;
 
-  constructor(public hammerProvider : HammerProvider, private renderer: Renderer2) {}
+  rippleEffectAnimationDuration: number = 300;
+
+  competencyState: ComptencyState;
+  subscription: Subscription;
+  faultCount: number;
+
+  constructor(
+    public hammerProvider : HammerProvider,
+    private renderer: Renderer2,
+    private store$: Store<StoreModel>,
+    ) {}
 
   ngOnInit() : void {
     this.hammerProvider.init(this.button);
     this.hammerProvider.addPressAndHoldEvent(this.recordFault);
+
+    this.competencyState = {
+      drivingFaultCount$: this.store$.pipe(
+        select(getCurrentTest),
+        select(getTestData),
+        select(testData => getDrivingFaultCount(testData, this.competency))),
+    };
+
+    const { drivingFaultCount$ } = this.competencyState;
+
+    const merged$ = merge(
+      drivingFaultCount$.pipe(map(count => this.faultCount = count)),
+    );
+
+    this.subscription = merged$.subscribe();
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
   }
+
+  getLabel = (): string => competencyLabels[this.competency];
+
   /**
    * Increments the fault count of the competency
    * @returns void
    */
   recordFault = (): void => {
-    // TODO - Dispatch ADD_FAULT Action Here
-    this.faultCount = this.faultCount + 1;
+    this.store$.dispatch(new AddDrivingFault({
+      competency: this.competency,
+      newFaultCount: this.faultCount ? this.faultCount + 1 : 1,
```

Comment:
How come we need to check for the existence of `this.faultCount`?

Reply:
When the selector first runs it returns null for faultCount in that instance we want newFaultCount to be 1